### PR TITLE
feat: attribute decorator

### PR DESCRIPTION
**Pull request by SoloJiang**
Docs: https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflecting-content-attributes-in-idl-attributes
 
The reflect rule is:

1. DOMTokenList type is not included in the cases, because developer cannot create domTokenList directly, more detail: https://dom.spec.whatwg.org/#domtokenlist

2. Attribute has two kinds: (1) Boolean attribute; (2) Common attribute

    - Boolean attribute:
      - property getter: if attribute value is null, it will return false, else return true;
                       if attribute value has been changed, it will return the changed boolean value
      - property setter: not reflect to attribute, more detail see InputElement checked property
      - property init:  the default value must be false,
                      and the return value depends on attribute value,
                      attribute value is null return false, else return true
    - Common attribute:
      - property getter: if attribute value has been changed, it will return the changed value,
                       else return attribute value
      - property setter: not reflect to attribute
      - property init: if attribute value is null, return default value; else return attribute value

### CODE REVIEW

File: packages/pwc/src/decorators/attribute/getter.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  const store = this._getReflectProperties().get(name);
+  const { isBoolean, attrName, initialValue } = store;
+
+  // Store value is highest weight
+  if (hasOwnProperty(store, 'value')) {
+    return store.value;
+  }
+
+  const attrValue = this.getAttribute(attrName);
+  if (attrValue === null && initialValue !== undefined) return initialValue;
```
Comment (cryzzchen) [inline packages/pwc/src/decorators/attribute/getter.ts:13]: 保持编程风格一致，这里包裹一下花括号

File: packages/pwc/src/decorators/attribute/getter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  return isBoolean || attrValue;
```
Comment (cryzzchen) [inline packages/pwc/src/decorators/attribute/getter.ts:15]: 这是不是意味着如果为布尔值，返回一定为 true?

File: packages/pwc/src/decorators/attribute/getter.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { hasOwnProperty } from '../../utils';
+
+export function attributeGetter(name: string) {
+  const store = this._getReflectProperties().get(name);
+  const { isBoolean, attrName, initialValue } = store;
+
+  // Store value is highest weight
+  if (hasOwnProperty(store, 'value')) {
```
Comment (cryzzchen) [inline packages/pwc/src/decorators/attribute/getter.ts:8]: store 可能不存在？

File: packages/pwc/src/decorators/attribute/getter.ts (hunk 2 above)
Comment (SoloJiang) [reply packages/pwc/src/decorators/attribute/getter.ts:15]: 是的，返回 false 的情况在上面考虑了

File: packages/pwc/src/decorators/attribute/getter.ts (hunk 3 above)
Comment (SoloJiang) [reply packages/pwc/src/decorators/attribute/getter.ts:8]: 是的，只有触发过 setter 才存在

Comment (SoloJiang) [reply packages/pwc/src/decorators/attribute/getter.ts:8]: store 一定存在

**Review by cryzzchen — APPROVED** [review]

File: packages/pwc/src/decorators/attribute/getter.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { hasOwnProperty } from '../../utils';
+
+export function attributeGetter(name: string) {
+  const store = this._getReflectProperties().get(name);
+  const { isBoolean, attrName, initialValue } = store;
+
+  // Store value is highest weight
```
Comment (ChrisCindy) [inline packages/pwc/src/decorators/attribute/getter.ts:7]:
```suggestion
  // Store value has highest weight
```

**Review by ChrisCindy — APPROVED** [review]

**Pull request merged**